osu.Game/Users/UserActivity.cs

```diff
@@ -87,22 +89,43 @@ public Editing(IBeatmapInfo info)
                 BeatmapInfo = info;
             }
 
-            public override string Status => @"Editing a beatmap";
+            public override string GetStatus(bool hideIdentifiableInformation = false) => @"Editing a beatmap";
         }
 
-        public class Spectating : UserActivity
+        public class Watching : UserActivity
         {
-            public override string Status => @"Spectating a game";
+            private readonly ScoreInfo score;
+
+            protected string Username => score.User.Username;
+
+            public BeatmapInfo BeatmapInfo => score.BeatmapInfo;
+
+            public Watching(ScoreInfo score)
+            {
+                this.score = score;
+            }
+
+            public override string GetStatus(bool hideIdentifiableInformation = false) => hideIdentifiableInformation ? @"Watching a replay" : $@"Watching {Username}'s replay";
```

Comment:
Maybe okay for now, but this probably should include more details than just the username.

Reply:
Agree, but since I took https://github.com/ppy/osu/pull/22529#discussion_r1097849032 into consideration I kept things simple. Maybe later we could show the ruleset without the `PlayingVerb`? (just the plain ruleset name)